docs/architecture/index.md

```diff
@@ -0,0 +1,130 @@
+
+A module refers to a **pluggable standalone component or extension** that provides additional functionality to the platform. Some of the ones supported by Zentral are shown in the following diagram.
+
+![Software Architecture Modules](/images/architecture/modules.svg)
+
+### Interactions with Endpoints
+
+#### To Zentral
+
+Some examples of Events and Status received from Endpoints  are:
+
+**Enrollment**: hostname, IP and other Endpoint identification.
+
+**Heartbeat**: CPU usage, memory usage, disk space among others. 
+
+**Security**: type of attack or event,  the affected system components, and any remediation actions taken.
+
+**Inventory**: installed applications, OS version, hardware specifications  and network configurations.
+
+#### From Zentral
+
+Some examples of Configurations, Assets, Activations and Managements sent to Endpoints  are:
+
+**Configuration/Assets**: updates and changes to security policies or software.
+
+**Security Alert**: blocking traffic from suspicious IP addresses or blocking access to certain network resources.
+
+**Management**: variety of purposes, such as initiating a system reboot, performing a network scan, or running a diagnostic tool.
+
+
+## Architecture decisions
+
+A brief description of the Architecture decisions, Frameworks and tools used in Zentral.
+
+**Python/Django Framework**: Zentral is built on top of the Python programming language and the Django web framework. This choice of technology provides a stable and well-supported foundation for the platform, as well as a wide range of third-party libraries and plugins that can be used to extend its functionality.
+
+**Modular Architecture**: designed as a modular platform, with separate components for data collection, processing, and storage. This modular architecture enables developers to add new features and capabilities to the platform without disrupting the core functionality.
+
+**Asynchronous Processing**: Zentral uses asynchronous processing techniques, such as Celery, RabbitMQ and Google PubSub to handle incoming data and alerts. This approach ensures that the platform can handle large volumes of data and respond quickly to security events.
+
+**Storage Flexibility**: the platform supports a range of different technologies to store events, including Elasticsearch, Kinesis, Splunk, Humio, OpenSearch and Azure. This flexibility enables organizations to choose the database technology that best fits their needs and requirements.
+
+**RESTful API**: Zentral provides a RESTful API that enables developers to build custom integrations and extensions, as well as to automate tasks and workflows within the platform (e.g. reporting).
+
+**Docker-Based Deployment**: The platform is designed to be deployed using [Docker](/deployment/docker-compose), a containerization technology that provides a standardized and reproducible way of deploying and managing the platform.  [Images on AWS and GCP: ZAIO (Zentral all in one)](/deployment/) are also provided.
```

Comment:
I would rather keep this docker thing in the Dev process part. Maybe a more general "Container based" deployment, because we always care about removing any friction for it.